### PR TITLE
Make unsafe lazy initialization safe again

### PR DESCRIPTION
**Pull request by Shredder121**
This uses the holder class pattern, so the JVM can do the locking when initializing for us.

### CODE REVIEW

**Comment by napstr** [comment]
Could you add this pattern for the SpotifyAPIWrapper class too please?

**Comment by napstr** [comment]
This is a nice pattern, the only caveat is that there must absolutely be no unhandled exception thrown during initialization.

One such thing could be happening, when the Ratelimiter constructor calls the constructor of the Blacklist, which in turn queries the database, which could be a source of rare but occasionally happening DatabaseNotReadyExceptions.

**Comment by Shredder121** [comment]
Good.. point, yeah.
So maybe just use double checked locking and storing it manually then?

**Comment by napstr** [comment]
Yep.

File: FredBoat/src/main/java/fredboat/util/rest/SpotifyAPIWrapper.java [hunk 1]
```diff
@@ -58,7 +58,6 @@
     private static final String URL_SPOTIFY_AUTHENTICATION_HOST = "https://accounts.spotify.com";
 
     private static final org.slf4j.Logger log = LoggerFactory.getLogger(SpotifyAPIWrapper.class);
-    private static volatile SpotifyAPIWrapper SPOTIFYAPIWRAPPER;
```
Comment (Shredder121) [inline FredBoat/src/main/java/fredboat/util/rest/SpotifyAPIWrapper.java]: @napstr I also removed this now.

**Pull request merged**